File: javascript/5_kyu/human_readable_time.js
// Human Readable Time
// https://www.codewars.com/kata/52685f7382004e774f0001f7/train/javascript

// Solution 1: While loop and Join
function humanReadable(time) {
    var hours = 0;
    var minutes = 0;
    var seconds = 0;

    while (time >= 3600) {
        hours++;
        time = time - 3600;
    }
    if (hours < 10) { hours = "0".concat(hours.toString())};

    while (time >= 60) {
        minutes++;
        time = time - 60;
    }
    if (minutes < 10) {minutes = '0'.concat(minutes.toString())};

    seconds = seconds + time;
    if (seconds < 10) {seconds = "0".concat(seconds.toString())};

    return [hours.toString(), minutes.toString(), seconds.toString()].join(":");
}


console.log(humanReadable(0));
console.log(humanReadable(5));
console.log(humanReadable(60));
console.log(humanReadable(86399));
console.log(humanReadable(359999));

File: python/5_kyu/human_readable_time.py
# Human Readable Time
# https://www.codewars.com/kata/52685f7382004e774f0001f7/train/python

# Solution 1: While loop and Join
def make_readable(time):
    hours, minutes, seconds = 0, 0, 0

    while time >= 3600:
        hours += 1
        time -= 3600
    
    if hours < 10:
        hours = "0" + str(hours)
    
    while time >= 60:
        minutes += 1
        time -= 60
    
    if minutes < 10:
        minutes = "0" + str(minutes)
    
    seconds = time
    if seconds < 10:
        seconds = "0" + str(seconds)


    return ":".join([str(hours), str(minutes), str(seconds)])
        

print(make_readable(0))
print(make_readable(5))
print(make_readable(60))
print(make_readable(86399))
print(make_readable(359999))